### DISCUSSION TITLE
sync with nextcloud podcast app

### DISCUSSION THREAD

**Discussion by thrillfall** [thrillfall/nextcloud-gpodder#23] [Ideas]
@jakobroehrl

Nice app, thanks!
It would be awesome, if the podcast app https://apps.nextcloud.com/apps/podcast would show these synced podcast sources, too.

**Comment by thrillfall** [comment]
the podcast app developer is aware of this app and will look into integration
https://github.com/AntennaPod/AntennaPod/pull/5243#issuecomment-870419935

**Reply by keunes** [reply]
And the relevant issue in the app's repo: https://git.project-insanity.org/onny/nextcloud-app-podcast/-/issues/103

**Comment by JonOfUs** [comment]
I experimented a bit with the podcast app and started implementing a simple synchronization with gpoddersync.
There I also noticed the EpisodeAction issue https://github.com/thrillfall/nextcloud-gpodder/issues/20.

The podcast app uses the [fyyd.de](https://fyyd.de) database and therefore each episode/podcast needs to be mapped to its fyyd-ID for identification. This can be done using the episode mp3 url, but this is error-prone (e.g. trackers in mp3 url) so it would be way better if it could use the episode GUID. So maybe this sync will only work good when AntennaPod changes to GUID for episode sync.

**Reply by thrillfall** [reply]
Antennapod is prepared for using the guid and with this change   https://github.com/thrillfall/AntennaPod/commit/30e074f6db051367e1fcd91e0a979e6ea8609ede the guide would be send.
But the PR at gpodder.net is still pending and this nc app is still in development

**Reply by JonOfUs** [reply]
Nice!
Yes, you're right. The podcast app is in development, too. It has not even reached v1.0, so there probably are many changes to come.

What do you think, how should this change be implemented in gpoddersync?
Should it just be a v2.0 that is no longer compatible with v1.0?
This will change a lot, e.g. the unique constraint violation usage probably needs to be reworked if backwards compability is wanted. Or a strict change to GUID as unique index and the deletion of all EpisodeActions without a GUID. But this definitely sounds like a lot of thinking.

**Reply by keunes** [reply]
> The podcast app is in development, too. It has not even reached v1.0, so there probably are many changes to come.

Unfortunately not much seems to have happened in terms of development the past months. It seems to me also that a lot of work needs to be done on the main app functionality before there might be an interest in synchronisation with other apps.

> Should it just be a v2.0 that is no longer compatible with v1.0?

That's a very important question, I'll copy it over to #91 so we can discuss in more detail there in a broader context.

**Comment by pklampros** [comment]
Apparently [the Music app now has podcast integration as well](https://github.com/owncloud/music/issues/786#issuecomment-907684995), just linking here for potential synchronisation.

**Comment by dugite-code** [comment]
FYI there is now the [repod](https://apps.nextcloud.com/apps/repod) app which uses this as a sync backend. Once Queue syncing is added to antennapod/nextcloud-gpodder this will be quite a compelling setup.
